moments/Parsing.py

```diff
@@ -0,0 +1,1149 @@
+        allele counts to the count of their occurences in the VCF file.
+    :rtype: dict
+    """
+    _clear_stats()
+
+    # Inspect `filters`
+    filters = _parse_filters(filters)
+
+    if bed_file:
+        regions, bed_chrom = _load_bed_file(bed_file)
+        mask = _bed_regions_to_mask(regions)
+        masked = True
+    else:
+        bed_chrom = None
+        masked = False
+
+    if interval is not None:
+        intervaled = True 
+    else:
+        intervaled = False
+
+    if anc_seq_file is not None and use_AA is True:
+        raise ValueError('You cannot use both `anc_seq_file` and `use_AA`')
+    if anc_seq_file:
+        ancestral_sequence = _load_fasta_file(anc_seq_file)
+        has_anc_seq = True
+    else:
+        if use_AA:
+            has_anc_seq = False
+        else:
+            has_anc_seq = False
+            print_out(current_time(),
+                'Polarizing with ``REF`` alleles: output SFS should be folded')
+    
+    if vcf_file.endswith('.gz'):
+        openfunc = gzip.open 
+    else:
+        openfunc = open
+
+    # Should the INFO field be inspected on each line?
+    if use_AA or 'INFO' in filters:
+        need_info = True
+    else:
+        need_info = False
+
+    tally = defaultdict(lambda: defaultdict(int))
+    vcf_chrom = None
+    counter = 0
+    
+    with openfunc(vcf_file, 'rb') as fin:
+        for lineb in fin:
+            line = lineb.decode()
+            if line.startswith('##'):
+                continue
+            if line.startswith('#CHROM'):
+                # Gather sample IDs
+                split_header = line.split()
+                sample_ids = split_header[9:]
+                if len(sample_ids) == 0:
+                    raise ValueError('VCF has zero samples')
+                if pop_file is not None and pop_mapping is not None:
+                    raise ValueError(
+                        'You cannot use both `pop_mapping` and `pop_file`'
+                    )
+                elif pop_file is not None:
+                    pop_mapping = _load_pop_file(pop_file)
+                elif pop_mapping is None:
+                    print_out(current_time(),
+                        'No populations given: placing all samples in "ALL"'
+                    )
+                    pop_mapping = {'ALL': sample_ids}
+                # Form a mapping from population IDs to sample indices
+                pop_idx = defaultdict(list)
+                for pop in pop_mapping:
+                    for sample_id in pop_mapping[pop]:
+                        if sample_id not in sample_ids:
+                            raise ValueError(
+                                f'Sample "{sample_id}" is absent from VCF'
+                            )
+                        pop_idx[pop].append(sample_ids.index(sample_id))
+                continue
+            
+            # Print a status report
+            if verbose > 1:
+                if counter % verbose == 0 and counter > 1:
```

Comment:
With verbose default set to 1, will this print every line? That would end up with a lot of printed lines, right?